fl4health/clients/basic_client.py

```diff
@@ -631,8 +657,6 @@ def validate(self) -> Tuple[float, Dict[str, Scalar]]:
         metrics = self.val_metric_manager.compute()
         self._handle_logging(loss_dict, metrics, is_validation=True)
 
-        # Checkpoint based on loss which is output of user defined compute_loss method
-        self._maybe_checkpoint(loss_dict["checkpoint"])
```

Comment:
Note that I moved this into the evaluate function rather than the validate function, as we may not want to checkpoint every time we validate etc.